### PR TITLE
IP Allow docs

### PR DESCRIPTION
**Pull request by danieltprice**
IP Allowlist documentation updates.

Previews:

- Main doc page:
https://neon-next-git-dpiece-ip-allowlist-neondatabase.vercel.app/docs/manage/projects#configure-ip-allow
- Security overview:
https://neon-next-git-dpiece-ip-allowlist-neondatabase.vercel.app/docs/security/security-overview#ip-allowlist-support
- Glossary entries:
https://neon-next-git-dpiece-ip-allowlist-neondatabase.vercel.app/docs/reference/glossary#cidr-notation
https://neon-next-git-dpiece-ip-allowlist-neondatabase.vercel.app/docs/reference/glossary#ip-allowlist
- Pro Plan docs:
https://neon-next-git-dpiece-ip-allowlist-neondatabase.vercel.app/docs/introduction/pro-plan#advanced-features
- Plans overview:
https://neon-next-git-dpiece-ip-allowlist-neondatabase.vercel.app/docs/introduction/plans#neon-plans

### CODE REVIEW

File: content/docs/introduction/pro-plan.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ The Neon Pro Plan comes with the following advanced features, and we plan to add
...
 - **Read replicas:** Neon's read replica feature enhances scalability and performance by allowing you to create read-only compute instances that operate on the same data as your read-write computes. With this feature, you can offload analytics or reporting queries, efficiently distribute read requests to achieve higher throughput, or provide read-only data access to certain users or applications. For more information, see [Read replicas](/docs/introduction/read-replicas).
+- **IP Allowlist:** Control access to the branch where your database resides. Any IP address not on the allowlist is automatically blocked, ensuring that only authorized users can access your data. The allowlist can be applied to all branches (the default) or to your [primary branch](#primary-branch) only. For more information, see [IP Allowlist](/docs/manage/projects#configure-the-ip-allowlist).
```
Comment (bgrenon) [inline content/docs/introduction/pro-plan.md:37]: Out of curiosity, why would you only apply it to the primary branch, but not other branches?

File: content/docs/manage/projects.md [hunk 2]
```diff
@@ -158,6 +159,47 @@ To configure the history retention period for a project:
 4. Use the slider to select the history retention period.
 5. Click **Save**.
 
+### Configure the IP Allowlist
+
+The IP Allowlist is a security feature avialable to [Neon Pro Plan](/docs/introduction/pro-plan) users that restricts access to the branch where your database resides to only those IP addresses that you specify, providing an added layer of security for your data. In Neon, the IP Allowlist is applied to all branches by default. Optionally, you can apply the IP Allowlist to your project's [primary branch](/docs/manage/branches#primary-branch) only. For instance, you might want to restrict access to the primary branch to a handful of trusted IPs while allowing broader access to development branches.
```
Comment (bgrenon) [inline content/docs/manage/projects.md:164]:
```suggestion
Available to [Neon Pro Plan](/docs/introduction/pro-plan) users, the IP Allowlist feature provides an added layer of security for your data, restricting access to the branch where your database resides to only those IP addresses that you specify. In Neon, the IP Allowlist is applied to all branches by default. Optionally, you can instead apply the IP Allowlist to your project's [primary branch](/docs/manage/branches#primary-branch) only. For instance, you might want to restrict access to the primary branch to a handful of trusted IPs while allowing broader access to development branches. 
```

**Review by bgrenon — APPROVED** [review]
Minor changes. All good!

**Review by Shridhad — APPROVED** [review]

File: content/docs/manage/projects.md [hunk 3]
```diff
@@ -173,6 +183,48 @@ To configure **IP Allow** for a project:
...
 5. Click **Apply changes**.
 
+</TabItem>
+
+<TabItem>
+
+The [Neon CLI projects command](https://neon.tech/docs/reference/cli-projects) supports IP Allow configuration. For example, this command configures IP for an existing Neon project:
+
+```bash
+projects update still-bird-22384887 --ip-allow 127.0.0.1 192.168.2.1/24 --ip-primary-only
```
Comment (Shridhad) [inline content/docs/manage/projects.md:193]:
```suggestion
neonctl projects update still-bird-22384887 --ip-allow 127.0.0.1 192.168.2.1/24 --ip-primary-only
```

File: content/docs/manage/projects.md [hunk 4]
```diff
@@ -158,6 +159,104 @@ To configure the history retention period for a project:
...
+</Tabs>
+
+
+#### How to specify IP addresses
+
+You can define an allowlist with individual IP addresses, IP ranges, or [CIDR notation](/docs/reference/glossary#cidr-notation). A combination of these options is also permitted. Multiple entries, whether they are the same or of different types, must be separated by a comma.
+
+- **Add individual IP addresses**: You can add individual IP addresses that you want to allow. This is useful for granting access to specific users or devices. This example represents a single IP address:
+
+  ```ini
```
Comment (evanshortiss) [inline content/docs/manage/projects.md:234]: This renders with the first two sections red. So `192.168` is red while the rest is white. Should we change from `ini` to just plaintext for this block?

File: content/docs/manage/projects.md [hunk 5]
```diff
@@ -158,6 +159,104 @@ To configure the history retention period for a project:
...
+
+- **Use CIDR notation**: For more advanced control, you can use [CIDR (Classless Inter-Domain Routing) notation](/docs/reference/glossary#cidr-notation). This is a compact way of defining a range of IPs and is useful for larger networks or subnets. Using CIDR notation can be advantageous when managing access to branches with numerous potential users, such as in a large development team or a company-wide network.
+
+  This CIDR notation example represents all 256 IP addresses from  `192.168.1.0` to `192.168.1.255`. 
+
+  ```ini
+  192.168.1.0/24
+  ```
+
+A combined example using all three options above, specified as a comma-separated list, would appear similar to the following:
```
Comment (evanshortiss) [inline content/docs/manage/projects.md:252]: Should we mention that white space is ignored?

File: content/docs/introduction/pro-plan.md [hunk 6]
```diff
@@ -34,6 +34,7 @@ The Neon Pro Plan comes with the following advanced features, and we plan to add
...
 - **Read replicas:** Neon's read replica feature enhances scalability and performance by allowing you to create read-only compute instances that operate on the same data as your read-write computes. With this feature, you can offload analytics or reporting queries, efficiently distribute read requests to achieve higher throughput, or provide read-only data access to certain users or applications. For more information, see [Read replicas](/docs/introduction/read-replicas).
+- **IP Allow:** Control access to the branch where your database resides. Any IP address not on the allowlist is automatically blocked, ensuring that only authorized users can access your data. The allowlist can be applied to all branches (the default) or to your [primary branch](#primary-branch) only. For more information, see [Configure IP Allow](/docs/manage/projects#configure-ip-allow).
```
Comment (evanshortiss) [inline content/docs/introduction/pro-plan.md:37]: "Users" or "networks"? The user might have valid credentials for authorisation, but the network they're attempting to connect from is the issue. It's possible that I'm being pedantic and should be ignored 😅

**Review by evanshortiss — APPROVED** [review]
Looks good. Only minor suggestions on my end.

**Review by lpetkov — APPROVED** [review]

File: content/docs/manage/projects.md [hunk 7]
```diff
@@ -158,6 +159,104 @@ To configure the history retention period for a project:
...
+5. Click **Apply changes**.
+
+</TabItem>
+
+<TabItem>
+
+The [Neon CLI projects command](https://neon.tech/docs/reference/cli-projects) supports IP Allow configuration. For example, this command configures IP for an existing Neon project:
+
+```bash
+neonctl projects update still-bird-22384887 --ip-allow 127.0.0.1 192.168.2.1/24 --ip-primary-only
```
Comment (vadim2404) [inline content/docs/manage/projects.md:193]: @Shridhad, did we release it?

File: content/docs/manage/projects.md [hunk 8]
```diff
@@ -158,6 +159,104 @@ To configure the history retention period for a project:
...
+     --header 'authorization: Bearer $NEON_API_KEY' \
+     --header 'content-type: application/json' \
+     --data '
+{
+  "project": {
+    "settings": {
+      "allowed_ips": {
+        "primary_branch_only": true,
+        "ips": [
+          "192.168.1.15, 192.168.1.16, 192.168.1.20-192.168.1.30, 192.168.1.0/24S"
```
Comment (vadim2404) [inline content/docs/manage/projects.md:215]: it should be an array of strings, shouldn't it?

File: content/docs/manage/projects.md (hunk 7 above)
Comment (Shridhad) [reply content/docs/manage/projects.md:193]: No, we did not. We are still holding the release as we have one discussion ongoing -  https://neondb.slack.com/archives/C036U0GRMRB/p1702912927587379 
cc: @lpetkov

File: content/docs/manage/projects.md (hunk 5 above)
Comment (danieltprice) [reply content/docs/manage/projects.md:252]: Added.

File: content/docs/introduction/pro-plan.md (hunk 6 above)
Comment (danieltprice) [reply content/docs/introduction/pro-plan.md:37]: Changed to IP addresses

File: content/docs/security/security-overview.md
[the remark this answers is not in the document]
Comment (danieltprice) [reply content/docs/security/security-overview.md:1]: changed

File: content/docs/manage/projects.md (hunk 8 above)
Comment (danieltprice) [reply content/docs/manage/projects.md]: removed for now

**Pull request merged**